2D3D_UnityProject/Assets/Scripts/PlayerController.cs

```diff
@@ -102,4 +156,15 @@ private void UpdateAnims(float xdir, float ydir, float moveType)
         anim.SetFloat("MoveY", ydir);
         anim.SetFloat("Speed", moveType);
     }
+
+    /// <summary>
+    /// called by OnTriggerEnter and OnTriggerExit of npc zones. Determines whether the player can talk or not
+    /// </summary>
+    /// <param name="withinZone">true on enter, false on exit</param>
+    /// <param name="partner">ID of npc</param>
+    public void SetInDialogueZone(bool withinZone, int partner)
+    {
+        inDialogueZone = withinZone;
+        dialoguePartner = partner;
+    }
```

Comment:
We really need to move this dialogue code into a separate component - Player class is getting too bloated